ecc_enclave/enclave/enclave.cpp

```diff
@@ -234,32 +246,101 @@ int ecall_cc_invoke(const char* encoded_args,
+
+        // set stream
+        istream =
+            pb_istream_from_buffer((const unsigned char*)decoded_crm.c_str(), decoded_crm.length());
+
+        b = pb_decode(&istream, fpc_ChaincodeRequestMessage_fields, &cc_request_message);
+        COND2LOGERR(!b, PB_GET_ERROR(&istream));
+        COND2LOGERR(cc_request_message.encrypted_request->size == 0, "zero size request");
+
+        // cc_request_message.encrypted_request->size);
+        ctx.encoded_args = (const char*)cc_request_message.encrypted_request->bytes;
+
+        // the dynamic memory in the message is release at the end
     }
 
-    if (ret != 0)
+    ctx.json_args = ctx.encoded_args;
+    ret = invoke(response, response_len, &response_len_out, &ctx);
+    COND2ERR(ret != 0);
+
+    b64_response = base64_encode((const unsigned char*)response, response_len_out);
+
     {
-        return SGX_ERROR_UNEXPECTED;
+        // TODO put response in protobuf and encode it
+
+        fpc_ChaincodeResponseMessage crm;
+        uint32_t stream_buffer_size = b64_chaincode_response_message_len_in / 4 * 3;
+        uint8_t stream_buffer[stream_buffer_size];
+        pb_ostream_t ostream;
+        std::string b64_crm_proto;
+
+        // create proto struct to encode
+        crm = {};
+        crm.encrypted_response = (pb_bytes_array_t*)pb_realloc(
+            crm.encrypted_response, PB_BYTES_ARRAY_T_ALLOCSIZE(b64_response.length()));
+        COND2LOGERR(crm.encrypted_response == NULL, "cannot allocate encrypted message");
+        crm.encrypted_response->size = b64_response.length();
+        ret = memcpy_s(crm.encrypted_response->bytes, crm.encrypted_response->size,
+            b64_response.c_str(), b64_response.length());
+        COND2LOGERR(ret != 0, "cannot encode field");
+
+        // encode proto
+        ostream = pb_ostream_from_buffer(stream_buffer, stream_buffer_size);
+        b = pb_encode(&ostream, fpc_ChaincodeResponseMessage_fields, &crm);
+        COND2LOGERR(!b, "error encoding proto");
+
+        pb_release(fpc_ChaincodeResponseMessage_fields, &crm);
+
+        // b64 encode
+        b64_crm_proto = base64_encode((const unsigned char*)stream_buffer, ostream.bytes_written);
+
+        // b64_response = base64_encode((const unsigned char*)response, response_len_out);
+
+        // ret = memcpy_s(b64_chaincode_response_message, b64_chaincode_response_message_len_in,
+        //    b64_response.c_str(), b64_response.length());
+
+        ret = memcpy_s(b64_chaincode_response_message, b64_chaincode_response_message_len_in,
+            b64_crm_proto.c_str(), b64_crm_proto.length());
+        COND2LOGERR(ret != 0, "cannot copy to response");
+
+        //*b64_chaincode_response_message_len_out = b64_response.length();
+        *b64_chaincode_response_message_len_out = b64_crm_proto.length();
     }
 
-    ret = gen_response("invoke", response, response_len_out, signature, &ctx);
+    // ret = gen_response("invoke", response, response_len_out, &ctx);
```

Comment:
```suggestion
     // TODO: generate signature (as short-cut for now over proposal _and_ args with consistency of proposal and args verified in "__endorse" rather than enclave)
    // ret = gen_response("invoke", response, response_len_out, &ctx);
```

Reply:
done